docs/Profiles/API-PROFILES-DESIGN.md

```diff
@@ -0,0 +1,2559 @@
+   - Thread-safe operations
+   - Cache size limits to prevent memory issues
+   - Separate cache regions for profiles vs. application mappings
+
+2. **Cache Configuration** (appsettings.json):
+
+   ```json
+   {
+     "Profiles": {
+       "Cache": {
+         "Enabled": true,
+         "ExpirationMinutes": 10,
+         "MaxCacheSize": 1000,
+         "PreloadProfiles": ["default", "student-read-only"],
+         "SlidingExpiration": true,
+         "ApplicationMappingCache": {
+           "Enabled": true,
+           "ExpirationMinutes": 15,
+           "MaxCacheSize": 5000
+         }
+       }
+     }
+   }
+   ```
+
+3. **Profile Resolution with Cache** (ProfileResolutionMiddleware.cs):
+   - For header-based: Check profile cache first before calling Config API
+   - For application-based: Check app-profiles cache first, then GET /v2/applications/{id}
+   - On cache miss: fetch from Config API and populate cache
+   - On cache hit: return cached profile object or application profiles
+   - Handle cache failures gracefully (fallback to Config API)
+   - Support both profile cache and application mapping cache
+
+**Considerations**:
+
+- **Eventual Consistency**: Profile and application assignment updates will have propagation delay (cache TTL)
+- **Cache Invalidation**: Consider adding cache invalidation API for immediate updates
+- **Application Cache**: Higher cache size needed due to more applications than profiles
+
+**Dependencies**:
+
+- Ticket 3 (Profile Resolution Infrastructure - includes application-based resolution)
+- Ticket 7a (Application Profile Assignment - provides application mappings)
+- Profile resolution middleware must be refactored to support dual caching (profiles + applications)
+
+**Component**: DMS
+
+**Reference**: [API Profiles Design Document](https://github.com/Ed-Fi-Alliance-OSS/Data-Management-Service/blob/main/docs/Profiles/API-PROFILES-DESIGN.md)
+
+**Priority**: Medium (Phase 2 - Performance optimization after core functionality)
+
+## Appendix B: XML Schema Definition
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema">
+  <xs:element name="Profile">
+    <xs:complexType>
+      <xs:sequence>
+        <xs:element name="Resource" maxOccurs="unbounded">
+          <xs:complexType>
+            <xs:sequence>
+              <xs:element name="ReadContentType" minOccurs="0">
+                <xs:complexType>
+                  <xs:sequence>
+                    <xs:element name="Property" maxOccurs="unbounded" minOccurs="0"/>
+                    <xs:element name="Collection" maxOccurs="unbounded" minOccurs="0"/>
+                    <xs:element name="Reference" maxOccurs="unbounded" minOccurs="0"/>
+                  </xs:sequence>
+                  <xs:attribute name="memberSelection" type="xs:string"/>
+                </xs:complexType>
+              </xs:element>
+              <xs:element name="WriteContentType" minOccurs="0">
```

Comment:
Is this intended to be complete? Both this and ReadContentType are missing a lot here.